src/VisualStudio/Core/Def/CodeCleanup/AbstractCodeCleanUpFixer.cs

```diff
@@ -193,11 +192,120 @@ private async Task<bool> FixAsync(
+                    }
+
+                    foreach (var projectChanges in solutionChanges.GetProjectChanges())
+                    {
+                        if (projectChanges.GetAddedAdditionalDocuments().Any()
+                            || projectChanges.GetRemovedAdditionalDocuments().Any()
+                            || projectChanges.GetAddedAnalyzerConfigDocuments().Any()
+                            || projectChanges.GetRemovedAnalyzerConfigDocuments().Any()
+                            || projectChanges.GetAddedAnalyzerReferences().Any()
+                            || projectChanges.GetRemovedAnalyzerReferences().Any()
+                            || projectChanges.GetAddedDocuments().Any()
+                            || projectChanges.GetRemovedDocuments().Any()
+                            || projectChanges.GetAddedMetadataReferences().Any()
+                            || projectChanges.GetRemovedMetadataReferences().Any()
+                            || projectChanges.GetAddedProjectReferences().Any()
+                            || projectChanges.GetRemovedProjectReferences().Any())
+                        {
+                            //something more than the document has changed. We can't apply this change.
+                            return false;
+                        }
+
+                        var changedDocuments = projectChanges.GetChangedDocuments()
+                            .Concat(projectChanges.GetChangedAdditionalDocuments())
+                            .Concat(projectChanges.GetChangedAnalyzerConfigDocuments()).ToImmutableArray();
+
+                        if (changedDocuments.Length == 0)
+                        {
+                            //nothing has changed in this project. We can't apply this change.
+                            return false;
+                        }
+
+                        foreach (var documentId in changedDocuments)
+                        {
+                            var originalDocument = projectChanges.NewProject.GetRequiredDocument(documentId);
+                            var currentDocument = currentSolution.GetTextDocument(documentId);
+
+                            if (currentDocument is null)
+                            {
+                                // Document is gone, we can't apply this change.
+
+                                return false;
+                            }
+
+                            if (originalDocument.HasTextChanged(currentDocument, ignoreUnchangeableDocument: false))
+                            {
+                                // Document has changed, we can't apply this change.
+                                return false;
+                            }
+                        }
+                    }
+
+                    // If we got here, all documents in this project are the same as they were when we started.
+                    var codeCleanUpChanges = solution.GetChanges(currentSolution);
```

Comment:
To get the changes that are in `changedSolution` we will be starting from `initialSolution`.

```suggestion
                    var codeCleanUpChanges = changedSolution.GetChanges(initialSolution);
```